diff -up macemu-master/BasiliskII/src/Unix/sys_unix.cpp~ macemu-master/BasiliskII/src/Unix/sys_unix.cpp
--- macemu-master/BasiliskII/src/Unix/sys_unix.cpp~	2013-03-04 19:14:20.000000000 +0100
+++ macemu-master/BasiliskII/src/Unix/sys_unix.cpp	2013-03-10 17:33:06.005230783 +0100
@@ -324,7 +324,7 @@ void SysAddDiskPrefs(void)
 			line[len-1] = 0;
 
 			// Parse line
-			char *dev, *mnt_point, *fstype;
+			char *dev = NULL, *mnt_point = NULL, *fstype = NULL;
 			if (sscanf(line, "%as %as %as", &dev, &mnt_point, &fstype) == 3) {
 				if (strcmp(fstype, "hfs") == 0)
 					PrefsAddString("disk", dev);
