diff -up macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/cxmon/src/mon.cpp~ macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/cxmon/src/mon.cpp
--- macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/cxmon/src/mon.cpp~	2016-03-22 10:27:20.000000000 +0100
+++ macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/cxmon/src/mon.cpp	2016-03-22 10:30:18.671721029 +0100
@@ -1112,7 +1112,7 @@ void mon_exit()
  *  Main function, read-execute loop
  */
 
-void mon(int argc, char **argv)
+void mon(int argc, const char **argv)
 {
 	bool done = false, interactive = true;
 
diff -up macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/cxmon/src/mon.h~ macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/cxmon/src/mon.h
--- macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/cxmon/src/mon.h~	2016-01-31 16:00:57.000000000 +0100
+++ macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/cxmon/src/mon.h	2016-03-22 10:30:19.137721654 +0100
@@ -30,7 +30,7 @@
 
 void mon_init();
 void mon_exit();
-void mon(int argc, char **argv);
+void mon(int argc, const char **argv);
 
 
 /*
diff -up macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/SheepShaver/src/kpx_cpu/src/cpu/ppc/ppc-execute.cpp~ macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/SheepShaver/src/kpx_cpu/src/cpu/ppc/ppc-execute.cpp
--- macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/SheepShaver/src/kpx_cpu/src/cpu/ppc/ppc-execute.cpp~	2016-01-31 16:00:57.000000000 +0100
+++ macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/SheepShaver/src/kpx_cpu/src/cpu/ppc/ppc-execute.cpp	2016-07-07 14:44:00.099249460 +0200
@@ -67,7 +67,7 @@ void powerpc_cpu::execute_illegal(uint32
 	disass_ppc(stdout, pc(), opcode);
 
 	// Start up mon in real-mode
-	char *arg[4] = {"mon", "-m", "-r", NULL};
+	const char *arg[4] = {"mon", "-m", "-r", NULL};
 	mon(3, arg);
 #endif
 	abort();
diff -up macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/SheepShaver/src/kpx_cpu/sheepshaver_glue.cpp~ macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/SheepShaver/src/kpx_cpu/sheepshaver_glue.cpp
--- macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/SheepShaver/src/kpx_cpu/sheepshaver_glue.cpp~	2016-01-31 16:00:57.000000000 +0100
+++ macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/SheepShaver/src/kpx_cpu/sheepshaver_glue.cpp	2016-07-07 14:45:33.741857603 +0200
@@ -83,7 +83,7 @@ static void enter_mon(void)
 {
 	// Start up mon in real-mode
 #if ENABLE_MON
-	char *arg[4] = {"mon", "-m", "-r", NULL};
+	const char *arg[4] = {"mon", "-m", "-r", NULL};
 	mon(3, arg);
 #endif
 }
diff -up macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/SheepShaver/src/kpx_cpu/src/cpu/ppc/ppc-cpu.cpp~ macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/SheepShaver/src/kpx_cpu/src/cpu/ppc/ppc-cpu.cpp
--- macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/SheepShaver/src/kpx_cpu/src/cpu/ppc/ppc-cpu.cpp~	2016-01-31 16:00:57.000000000 +0100
+++ macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/SheepShaver/src/kpx_cpu/src/cpu/ppc/ppc-cpu.cpp	2016-07-07 14:48:51.031138859 +0200
@@ -527,7 +527,7 @@ bool powerpc_cpu::check_spcflags()
 		spcflags().clear(SPCFLAG_CPU_ENTER_MON);
 #if ENABLE_MON
 		// Start up mon in real-mode
-		char *arg[] = {
+		const char *arg[] = {
 			"mon",
 #ifdef SHEEPSHAVER
 			"-m",
diff -up macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/SheepShaver/src/kpx_cpu/src/cpu/ppc/ppc-translate.cpp~ macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/SheepShaver/src/kpx_cpu/src/cpu/ppc/ppc-translate.cpp
--- macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/SheepShaver/src/kpx_cpu/src/cpu/ppc/ppc-translate.cpp~	2016-01-31 16:00:57.000000000 +0100
+++ macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/SheepShaver/src/kpx_cpu/src/cpu/ppc/ppc-translate.cpp	2016-07-07 14:48:12.805890613 +0200
@@ -95,7 +95,7 @@ static void disasm_block(int target, uin
 			target == TARGET_POWERPC ? "d" : "x",
 			start, start + length - 1);
 
-	char *arg[] = {"mon",
+	const char *arg[] = {"mon",
 #ifdef SHEEPSHAVER
 				   "-m",
 #endif
--- macemu-e273bb1a0b4f6e35bcdbf6cf918aa0ca3e6d99da/BasiliskII/src/emul_op.cpp~	2017-09-04 20:56:34.000000000 +0200
+++ macemu-e273bb1a0b4f6e35bcdbf6cf918aa0ca3e6d99da/BasiliskII/src/emul_op.cpp	2017-10-01 18:12:37.352082375 +0200
@@ -70,7 +70,7 @@ void EmulOp(uint16 opcode, M68kRegisters
 				   r->sr);
 			VideoQuitFullScreen();
 #ifdef ENABLE_MON
-			char *arg[4] = {"mon", "-m", "-r", NULL};
+			const char *arg[4] = {"mon", "-m", "-r", NULL};
 			mon(3, arg);
 #endif
 			QuitEmulator();
@@ -575,7 +575,7 @@ void EmulOp(uint16 opcode, M68kRegisters
 				   r->a[0], r->a[1], r->a[2], r->a[3], r->a[4], r->a[5], r->a[6], r->a[7],
 				   r->sr);
 #ifdef ENABLE_MON
-			char *arg[4] = {"mon", "-m", "-r", NULL};
+			const char *arg[4] = {"mon", "-m", "-r", NULL};
 			mon(3, arg);
 #endif
 			break;
@@ -592,7 +592,7 @@ void EmulOp(uint16 opcode, M68kRegisters
 				   r->a[0], r->a[1], r->a[2], r->a[3], r->a[4], r->a[5], r->a[6], r->a[7],
 				   r->sr);
 #ifdef ENABLE_MON
-			char *arg[4] = {"mon", "-m", "-r", NULL};
+			const char *arg[4] = {"mon", "-m", "-r", NULL};
 			mon(3, arg);
 #endif
 			QuitEmulator();
--- macemu-e273bb1a0b4f6e35bcdbf6cf918aa0ca3e6d99da/BasiliskII/src/uae_cpu/newcpu.h~	2017-09-04 20:56:34.000000000 +0200
+++ macemu-e273bb1a0b4f6e35bcdbf6cf918aa0ca3e6d99da/BasiliskII/src/uae_cpu/newcpu.h	2017-10-01 18:15:57.469639476 +0200
@@ -206,7 +206,7 @@ static __inline__ void m68k_setpc (uaecp
 	if (IS_BREAK_POINT(newpc)) {
 		printf("Stopped at break point address: %08lx. Last PC: %08lx\n", newpc, previous_pc);
 		m68k_dumpstate(NULL);
-		char *arg[4] = {"mon", "-m", "-r", NULL};
+		const char *arg[4] = {"mon", "-m", "-r", NULL};
 		mon(3, arg);
 	}
 #endif // end of #if ENABLE_MON
@@ -223,7 +223,7 @@ static __inline__ void m68k_incpc (uae_s
 	if (IS_BREAK_POINT(next_pc)) {
 		printf("Stopped at break point address: %08lx. Last PC: %08lx\n", next_pc, previous_pc);
 		m68k_dumpstate(NULL);
-		char *arg[4] = {"mon", "-m", "-r", NULL};
+		const char *arg[4] = {"mon", "-m", "-r", NULL};
 		mon(3, arg);
 	}
 #endif // end of #if ENABLE_MON
--- macemu-e273bb1a0b4f6e35bcdbf6cf918aa0ca3e6d99da/BasiliskII/src/Unix/main_unix.cpp~	2017-09-04 20:56:34.000000000 +0200
+++ macemu-e273bb1a0b4f6e35bcdbf6cf918aa0ca3e6d99da/BasiliskII/src/Unix/main_unix.cpp	2017-10-01 18:17:56.182563173 +0200
@@ -291,7 +291,7 @@ static void sigsegv_dump_state(sigsegv_i
 #endif
 	VideoQuitFullScreen();
 #ifdef ENABLE_MON
-	char *arg[4] = {"mon", "-m", "-r", NULL};
+	const char *arg[4] = {"mon", "-m", "-r", NULL};
 	mon(3, arg);
 #endif
 	QuitEmulator();
@@ -948,7 +948,7 @@ static void sigint_handler(...)
 	m68k_dumpstate(&nextpc);
 #endif
 	VideoQuitFullScreen();
-	char *arg[4] = {"mon", "-m", "-r", NULL};
+	const char *arg[4] = {"mon", "-m", "-r", NULL};
 	mon(3, arg);
 	QuitEmulator();
 }
@@ -1495,7 +1495,7 @@ ill:		printf("SIGILL num %d, code %d\n",
 
 			VideoQuitFullScreen();
 #ifdef ENABLE_MON
-			char *arg[4] = {"mon", "-m", "-r", NULL};
+			const char *arg[4] = {"mon", "-m", "-r", NULL};
 			mon(3, arg);
 #endif
 			QuitEmulator();
--- macemu-e273bb1a0b4f6e35bcdbf6cf918aa0ca3e6d99da/SheepShaver/src/Unix/main_unix.cpp~	2017-09-04 20:56:34.000000000 +0200
+++ macemu-e273bb1a0b4f6e35bcdbf6cf918aa0ca3e6d99da/SheepShaver/src/Unix/main_unix.cpp	2017-10-01 18:18:04.246625919 +0200
@@ -1402,7 +1402,7 @@ static void *tick_func(void *arg)
 #ifdef ENABLE_MON
 			// Start up mon in real-mode
 			printf("Welcome to the sheep factory.\n");
-			char *arg[4] = {"mon", "-m", "-r", NULL};
+			const char *arg[4] = {"mon", "-m", "-r", NULL};
 			mon(3, arg);
 #endif
 			return NULL;
