diff -up macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/BasiliskII/src/Unix/configure.ac~ macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/BasiliskII/src/Unix/configure.ac
--- macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/BasiliskII/src/Unix/configure.ac~	2016-01-31 16:00:57.000000000 +0100
+++ macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/BasiliskII/src/Unix/configure.ac	2016-03-22 10:26:00.590374903 +0100
@@ -147,11 +147,11 @@ dnl We use mon if possible.
 MONSRCS=
 if [[ "x$WANT_MON" = "xyes" ]]; then
   AC_MSG_CHECKING(for mon)
-  mon_srcdir=../../../mon/src
+  mon_srcdir=../../../cxmon/src
   if grep mon_init $mon_srcdir/mon.h >/dev/null 2>/dev/null; then
     AC_MSG_RESULT(yes)
     AC_DEFINE(ENABLE_MON, 1, [Define if using "mon".])
-    MONSRCS="$mon_srcdir/mon.cpp $mon_srcdir/mon_6502.cpp $mon_srcdir/mon_z80.cpp $mon_srcdir/mon_cmd.cpp $mon_srcdir/mon_disass.cpp $mon_srcdir/mon_ppc.cpp $mon_srcdir/mon_lowmem.cpp $mon_srcdir/disass/floatformat.c $mon_srcdir/disass/i386-dis.c $mon_srcdir/disass/m68k-dis.c $mon_srcdir/disass/m68k-opc.c $mon_srcdir/disass/mips-dis.c $mon_srcdir/disass/mips-opc.c $mon_srcdir/disass/mips16-opc.c"
+    MONSRCS="$mon_srcdir/mon.cpp $mon_srcdir/mon_6502.cpp $mon_srcdir/mon_z80.cpp $mon_srcdir/mon_cmd.cpp $mon_srcdir/mon_disass.cpp $mon_srcdir/mon_ppc.cpp $mon_srcdir/mon_lowmem.cpp $mon_srcdir/disass/floatformat.c $mon_srcdir/disass/i386-dis.c $mon_srcdir/disass/m68k-dis.c $mon_srcdir/disass/m68k-opc.c"
     CXXFLAGS="$CXXFLAGS -I$mon_srcdir -I$mon_srcdir/disass"
     AC_CHECK_LIB(ncurses, tgetent, ,
       [AC_CHECK_LIB(termcap, tgetent, ,
diff -up macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/SheepShaver/src/Unix/configure.ac~ macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/SheepShaver/src/Unix/configure.ac
--- macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/SheepShaver/src/Unix/configure.ac~	2016-01-31 16:00:57.000000000 +0100
+++ macemu-b58a9260bd1422a28e4c0b7b6bb71d26603bc3e1/SheepShaver/src/Unix/configure.ac	2016-07-07 14:40:04.682720595 +0200
@@ -114,7 +114,7 @@ x/* | x.*)
   WANT_MON=yes
   ;;
 xyes)
-  mon_srcdir=../../../mon/src
+  mon_srcdir=../../../cxmon/src
   ;;
 esac
 if [[ "x$WANT_MON" = "xyes" ]]; then
@@ -122,7 +122,7 @@ if [[ "x$WANT_MON" = "xyes" ]]; then
   if grep mon_init $mon_srcdir/mon.h >/dev/null 2>/dev/null; then
     AC_MSG_RESULT(yes)
     AC_DEFINE(ENABLE_MON, 1, [Define if using "mon".])
-    MONSRCS="$mon_srcdir/mon.cpp $mon_srcdir/mon_6502.cpp $mon_srcdir/mon_z80.cpp $mon_srcdir/mon_cmd.cpp $mon_srcdir/mon_lowmem.cpp $mon_srcdir/mon_disass.cpp $mon_srcdir/mon_ppc.cpp $mon_srcdir/disass/floatformat.c $mon_srcdir/disass/i386-dis.c $mon_srcdir/disass/m68k-dis.c $mon_srcdir/disass/m68k-opc.c $mon_srcdir/disass/mips-dis.c $mon_srcdir/disass/mips-opc.c $mon_srcdir/disass/mips16-opc.c"
+    MONSRCS="$mon_srcdir/mon.cpp $mon_srcdir/mon_6502.cpp $mon_srcdir/mon_z80.cpp $mon_srcdir/mon_cmd.cpp $mon_srcdir/mon_lowmem.cpp $mon_srcdir/mon_disass.cpp $mon_srcdir/mon_ppc.cpp $mon_srcdir/disass/floatformat.c $mon_srcdir/disass/i386-dis.c $mon_srcdir/disass/m68k-dis.c $mon_srcdir/disass/m68k-opc.c"
     CXXFLAGS="$CXXFLAGS -I$mon_srcdir -I$mon_srcdir/disass"
     AC_CHECK_LIB(ncurses, tgetent, ,
       [AC_CHECK_LIB(termcap, tgetent, ,
