diff -up macemu-master/SheepShaver/src/Unix/configure.ac~ macemu-master/SheepShaver/src/Unix/configure.ac
--- macemu-master/SheepShaver/src/Unix/configure.ac~	2013-03-04 19:14:20.000000000 +0100
+++ macemu-master/SheepShaver/src/Unix/configure.ac	2013-03-12 13:02:00.559524476 +0100
@@ -652,7 +652,7 @@ esac
 dnl BINCUE
 AS_IF([test  "x$with_bincue" = "xyes" ], [have_bincue=yes], [have_bincue=no])
 AS_IF([test  "x$have_bincue" = "xyes" ], [
-   if [[ "xOSX_CORE_AUDIO" = "xno"  -a "x$WANT_SDL_AUDIO"="xno"]]; then
+   if [[ "xOSX_CORE_AUDIO" = "xno"  -a "x$WANT_SDL_AUDIO"="xno" ]]; then
        AC_MSG_ERROR([You need SDL or OSX Core Audio to use BINCUE support.])
    else
        CPPFLAGS="$CPPFLAGS -DBINCUE $OSX_CORE_AUDIO"
