diff -up SheepShaver-2.3/src/Unix/Makefile.in.nostrip SheepShaver-2.3/src/Unix/Makefile.in
--- SheepShaver-2.3/src/Unix/Makefile.in.nostrip	2006-05-08 19:19:45.000000000 +0200
+++ SheepShaver-2.3/src/Unix/Makefile.in	2008-10-18 21:37:07.000000000 +0200
@@ -35,7 +35,7 @@ BLESS = @BLESS@
 EXEEXT = @EXEEXT@
 KEYCODES = @KEYCODES@
 INSTALL = @INSTALL@
-INSTALL_PROGRAM = @INSTALL_PROGRAM@ -s
+INSTALL_PROGRAM = @INSTALL_PROGRAM@
 INSTALL_DATA = @INSTALL_DATA@
 
 SLIRP_CFLAGS = @SLIRP_CFLAGS@
