diff -up macemu-master/SheepShaver/src/Unix/Makefile.in~ macemu-master/SheepShaver/src/Unix/Makefile.in
--- macemu-master/SheepShaver/src/Unix/Makefile.in~	2013-03-04 19:14:20.000000000 +0100
+++ macemu-master/SheepShaver/src/Unix/Makefile.in	2013-03-10 21:37:07.317286985 +0100
@@ -35,7 +35,7 @@ BLESS = @BLESS@
 EXEEXT = @EXEEXT@
 KEYCODES = @KEYCODES@
 INSTALL = @INSTALL@
-INSTALL_PROGRAM = @INSTALL_PROGRAM@ -s
+INSTALL_PROGRAM = @INSTALL_PROGRAM@
 INSTALL_DATA = @INSTALL_DATA@
 
 SLIRP_CFLAGS = @SLIRP_CFLAGS@
